eng/common/TestResources/Remove-TestResources.ps1

```diff
@@ -124,7 +124,13 @@ if (![string]::IsNullOrWhiteSpace($ServiceDirectory)) {
     $preRemovalScript = Join-Path -Path $root -ChildPath 'remove-test-resources-pre.ps1'
     if (Test-Path $preRemovalScript) {
         Log "Invoking pre resource removal script '$preRemovalScript'"
-        &$preRemovalScript -ResourceGroupName $ResourceGroupName @PSBoundParameters
+        if ($BaseName){
```

Comment:
Shouldn't this be based on ResourceGroupName?

Reply:
by the time we get here, ResourceGroup has already been calculated and have a persisted value.

Reply:
I guess I don't understand the logic here, why if BaseName is set do we include ResourceGroupName? Should we instead make some other condition above that makes it a little clearer why this condition is interesting for this parameter?

Reply:
That is a fair assessment. If ResourceGroupName is not provided, we construct it using the BaseName, so I can remove the condition on the BaseName and base it on whether or not we have overriden the ResourceGroupName or not. 
Something like this:
```

$isRgParamProvided = $true

if (!$ResourceGroupName) {
    # Format the resource group name like in New-TestResources.ps1.
    $isRgParamProvided = $false
    $ResourceGroupName = "rg-$BaseName"
}

if (![string]::IsNullOrWhiteSpace($ServiceDirectory)) {
    $root = [System.IO.Path]::Combine("$PSScriptRoot/../../../sdk", $ServiceDirectory) | Resolve-Path
    $preRemovalScript = Join-Path -Path $root -ChildPath 'remove-test-resources-pre.ps1'
    if (Test-Path $preRemovalScript) {
        Log "Invoking pre resource removal script '$preRemovalScript'"
        if ($isRgParamProvided){
            &$preRemovalScript @PSBoundParameters
        }
        else {
            &$preRemovalScript -ResourceGroupName $ResourceGroupName @PSBoundParameters
        }
        
    }
}

```

Reply:
I will do what Heath just suggested ! that seems like the best approach.

Reply:
You should use the `$PSCmdlet.ParameterSetName` if you want to know whether a required parameter for a parameter set (they can and do belong to multiple, so it depends on which). This will make the script easier to maintain over time. `ResourceGroupName` and `BaseName` can't be specified at the same time, and the parameter group will tell you that.

Reply:
For example,

```powershell
if ($PSCmdlet.ParameterSetName.StartsWith('ResourceGroup')) {
  $PSBoundParameters.Add('ResourceGroupName', $ResourceGroupName);
}

&$preRemovalScript @PSBoundParameters
```